daemon/daemon_unix.go

```diff
@@ -342,6 +353,21 @@ func verifyContainerResources(resources *containertypes.Resources, sysInfo *sysi
 	}
 
 	// cpu subsystem checks and adjustments
+	if resources.NanoCPUs > 0 && resources.CPUPeriod > 0 {
+		return warnings, fmt.Errorf("Conflicting options: Nano CPUs and CPU Period cannot both be set")
+	}
+	if resources.NanoCPUs > 0 && resources.CPUQuota > 0 {
+		return warnings, fmt.Errorf("Conflicting options: Nano CPUs and CPU Quota cannot both be set")
+	}
+	if resources.NanoCPUs > 0 && (!sysInfo.CPUCfsPeriod || !sysInfo.CPUCfsQuota) {
+		warnings = append(warnings, "Your kernel does not support CPU cfs period/quota or the cgroup is not mounted. NanoCPUs discarded.")
+		logrus.Warn("Your kernel does not support CPU cfs period/quota or the cgroup is not mounted. NanoCPUs discarded.")
```

Comment:
I know for other options we warn, but I feel like we should error out when we can't set this.